mlir-tensorrt/compiler/include/mlir-tensorrt/Compiler/OptionsProviders.h

```diff
@@ -57,19 +65,57 @@ struct DebugOptions {
   }
 };
 
-struct ExecutorOptions {
+struct ExecutorOptions : public OptionsProvider {
   /// The host index bit-width.
   int64_t executorIndexBitwidth{64};
 
   /// Whether to pass memref's as struct/table in function calls.
   bool executorUsePackedMemRefCConv{true};
 
-  void addToOptions(mlir::OptionsContext &context) {
+  void addToOptions(mlir::OptionsContext &context) override {
     context.addOption("executor-index-bitwidth", executorIndexBitwidth,
                       llvm::cl::init(64));
   }
 };
 
+struct DeviceOptions : public OptionsProvider {
+  /// Target device compute capability (SM version)
+  int64_t computeCapability;
+
+  /// Target device max shared memory per block (kilobytes)
+  int64_t maxSharedMemoryPerBlockKb;
+
+  /// Target device maximum 4-byte register sper block.
+  uint64_t maxRegistersPerBlock;
+
+  void addToOptions(mlir::OptionsContext &context) override {
+    context.addOption(
+        "device-compute-capability", computeCapability, llvm::cl::init(64),
+        llvm::cl::desc("Sets the device compute capbility. Only relevant "
+                       "if '--device-infer-from-host=false'"));
+    context.addOption("device-max-shared-memory-per-block-kb",
+                      maxSharedMemoryPerBlockKb, llvm::cl::init(0));
+    context.addOption("device-max-registers-per-block", maxRegistersPerBlock,
+                      llvm::cl::init(0));
+    context.addOption("device-infer-from-host", shouldInferFromHost,
+                      llvm::cl::init(true),
+                      llvm::cl::desc("Infers device information from host"));
+  }
+
+  llvm::Error finalize() override;
+
+  // TODO (pranavm): This should ideally be private but needs to be set from
```

Comment:
Need to figure out if I can make this option private if we are able to unify `DeviceOptions` and `ClusteringPipelineCliOpts`.